### PR TITLE
Add a set_config command to the cluster_tool

### PR DESCRIPTION
**Pull request by AlCutter**

### CODE REVIEW

File: cpp/tools/clustertool.h [hunk 1]
```diff
@@ -10,9 +18,15 @@ class ClusterTool {
...
   //  - Creates the /cluster_config entry.
-  static util::Status InitLog(TreeSigner<Logged>* tree_signer,
+  static util::Status InitLog(const ct::ClusterConfig& cluster_config,
+                              TreeSigner<Logged>* tree_signer,
                               ConsistentStore<Logged>* consistent_store);
 
+  // Sets the cluster config
+  static util::Status SetClusterConfig(
+      const ct::ClusterConfig& cluster_config,
+      ConsistentStore<Logged>* consistent_store);
```
Comment (pphaneuf) [inline cpp/tools/clustertool.h]: I was already feeling that in the previous pull request: either those should be free functions, or that class should start pulling its weight a bit and have something (like the tree signer and the consistent store)? Even then, that wouldn't really be state, just all const stuff, so I'd probably favour free functions? But I'll leave it to you...

File: cpp/tools/clustertool_main.cc [hunk 2]
```diff
@@ -47,64 +66,121 @@ void Usage() {
...
+            << "  initlog     Initialise a new log.\n"
+            << "  set_config  Set/Change a cluster's config.\n";
 }
 
 
+unique_ptr<LogSigner> BuildLogSigner() {
+  CHECK(!FLAGS_key.empty());
+  util::StatusOr<EVP_PKEY*> pkey(ReadPrivateKey(FLAGS_key));
+  CHECK_EQ(pkey.status(), util::Status::OK);
+  return unique_ptr<LogSigner>(new LogSigner(pkey.ValueOrDie()));
```
Comment (pphaneuf) [inline cpp/tools/clustertool_main.cc]: I really wish there was a ```make_unique<LogSigner>(pkey.ValueOrDie())```, this is some bad case of stuttering... The "solution" is Go, I suppose? :wink:

**Comment by pphaneuf** [comment]
LGTM

File: cpp/tools/clustertool.h (hunk 1 above)
Comment (AlCutter) [inline cpp/tools/clustertool.h]: Yeah, fair point - I've removed their shackles and they're free to gambol through the `.text SECTION` as they please.

**Comment by pphaneuf** [comment]
Looks even gooder to me!

**Pull request merged**